content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Optimize queries
+description: >
+  Optimize your SQL and InfluxQL queries to improve performance and reduce their memory and compute (CPU) requirements.
+weight: 401
+menu:
+  influxdb_cloud_serverless:
+    name: Optimize queries
+    parent: Execute queries
+influxdb/cloud-serverless/tags: [query, sql, influxql]
+related:
+  - /influxdb/cloud-serverless/query-data/sql/
+  - /influxdb/cloud-serverless/query-data/influxql/
+  - /influxdb/cloud-serverless/query-data/execute-queries/troubleshoot/
+  - /influxdb/cloud-serverless/reference/client-libraries/v3/
+---
+
+## Troubleshoot query performance
+
+Use the following tools to help you identify performance bottlenecks and troubleshoot problems in queries:
+
+<!-- TOC -->
+
+- [Troubleshoot query performance](#troubleshoot-query-performance)
+  - [EXPLAIN and ANALYZE](#explain-and-analyze)
+  - [Enable trace logging](#enable-trace-logging)
+
+<!-- /TOC -->
+
+### EXPLAIN and ANALYZE
+
+To view the query engine's execution plan and metrics for an SQL query, prepend [`EXPLAIN`](/influxdb/cloud-serverless/reference/sql/explain/) or [`EXPLAIN ANALYZE`](/influxdb/cloud-serverless/reference/sql/explain/#explain-analyze) to the query.
+The report can reveal query bottlenecks such as a large number of table scans or parquet files, and can help triage the question, "Is the query slow due to the amount of work required or due to a problem with the schema, compactor, etc.?"
+
+The following example shows how to use the InfluxDB v3 Python client library and pandas to view `EXPLAIN` and `EXPLAIN ANALYZE` results for a query:
+
+<!-- Import for tests and hide from users.
+```python
+import os
+```
+-->
+<!--pytest-codeblocks:cont-->
+{{% code-placeholders "BUCKET_NAME|API_TOKEN|APP_REQUEST_ID" %}}
+```python
+from influxdb_client_3 import InfluxDBClient3
+import pandas as pd
+import tabulate # Required for pandas.to_markdown()
+
+def explain_and_analyze():
+  print('Use SQL EXPLAIN and ANALYZE to view query plan information.')
+
+  # Instantiate an InfluxDB client.
+  client = InfluxDBClient3(token = f"API_TOKEN",
+                          host = f"{{< influxdb/host >}}",
+                          database = f"BUCKET_NAME")
+
+  sql_explain = '''EXPLAIN SELECT *
+        FROM home
+        WHERE time >= now() - INTERVAL '90 days'
+        ORDER BY time'''
+
+  table = client.query(sql_explain)
+  df = table.to_pandas()
+
+  sql_explain_analyze = '''EXPLAIN ANALYZE SELECT *
+      FROM home
+      WHERE time >= now() - INTERVAL '90 days'
+      ORDER BY time'''
+
+  table = client.query(sql_explain_analyze)
+
+  # Combine the Dataframes and output the plan information.
+  df = pd.concat([df, table.to_pandas()])
```

Comment:
The information shown by `EXPLAIN ANALYZE` should be a superset of a plain `EXPLAIN` query. There's no need to combine both. Either use one or the other.